packages/webview_flutter/lib/src/webview_method_channel.dart

```diff
@@ -122,6 +127,7 @@ class MethodChannelWebViewPlatform implements WebViewPlatformController {
     _addIfNonNull('jsMode', settings.javascriptMode?.index);
     _addIfNonNull('hasNavigationDelegate', settings.hasNavigationDelegate);
     _addIfNonNull('debuggingEnabled', settings.debuggingEnabled);
+    _addIfNonNull('userAgent', settings.userAgent);
```

Comment:
What happens if we rebuild a WebView that had an explicitly set `userAgent` and set the `userAgent` to null. IIUC the way this currently propagates will lead to preserving the previous user agent. I'd expect it to go back to the platform's default user agent.

We should add an test for this scenario as well (build with an explicitly set user agent, rebuild with user agent set to null)

Reply:
Makes sense. I changed it so that the user agent can be set to null and the default platform user agent is used after a rebuild. Additionally I adapted the same logic for the FakePlatformWebView unit tests.

Reply:
Thanks!
I think we should avoid updating the user agent each time the widget is rebuilt.
It seems like we're lacking the ability to express presence/absence of a setting that may be null. One way to do it would be to use some kind of an optional type.

I pushed https://github.com/flutter/plugins/pull/1920/commits/79761515b33c3e96234e7b0555651c74f971adee with a proposal on how to do that, let me know whether that makes sense to you.

Reply:
I think that is a nice way to handle nullable settings and I agree that updating the user agent every rebuild is unnecessary.